src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    let shouldInclude = true;
+
+    for (const animal of Animal.alive) {
+      if (animal.name === this.name) {
+        shouldInclude = false;
+      }
+    }
+
+    if (shouldInclude === true) {
+      Animal.alive.push(this);
+    }
+    Animal.alive.sort((a, b) => b.health - a.health);
```

Comment:
I understand why you've used `sort` here – to make sure `pop` works fine
but it could have been done in a more simple way – just by using `filter` method